v21.1/alembic.md

```diff
@@ -0,0 +1,569 @@
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ git clone git@github.com:cockroachlabs/simple-crud-python-sqlalchemy.git
+    ~~~
+
+## Step 3. Install and initialize Alembic
+
+1. Navigate to the `simple-crud-python-sqlalchemy` project directory, and run the following commands to create and start a virtual environment:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ python3 -m venv env
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ source env/bin/activate
+    ~~~
+
+1. Install the `alembic`, [`sqlalchemy-cockroachdb`](https://github.com/cockroachdb/sqlalchemy-cockroachdb), and [`psycopg2`](https://github.com/psycopg/psycopg2/) modules to the virtual environment:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ pip install sqlalchemy-cockroachdb psycopg2 alembic
+    ~~~
+
+    The `sqlalchemy-cockroachdb` and `psycopg2` modules are required to use the CockroachDB adapter that the app uses to run transactions against a CockroachDB cluster.
+
+    `alembic` includes the `sqlalchemy` module, which is a primary dependency of the `simple-crud-python-sqlalchemy` sample app. The `alembic` install also includes the `alembic` command line tool, which we use throughout the tutorial to manage migrations.
+
+1. Use the `alembic` command-line tool to initialize Alembic for the project:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ alembic init alembic
+    ~~~
+
+    ~~~
+    Creating directory /path/simple-crud-python-
+    sqlalchemy/alembic ...  done
+    Creating directory /path/simple-crud-python-
+    sqlalchemy/alembic/versions ...  done
+    Generating /path/simple-crud-python-
+    sqlalchemy/alembic/script.py.mako ...  done
+    Generating /path/simple-crud-python-
+    sqlalchemy/alembic/env.py ...  done
+    Generating /path/simple-crud-python-
+    sqlalchemy/alembic/README ...  done
+    Generating /path/simple-crud-python-
+    sqlalchemy/alembic.ini ...  done
+    Please edit configuration/connection/logging settings in
+    '/path/simple-crud-python-sqlalchemy/alembic.ini' before
+    proceeding.
+    ~~~
+
+    This command creates a migrations directory called `alembic`. This directory will contain the files that specify the schema migrations for the app.
+
+    The command also creates a properties file called `alembic.ini` at the top of the project directory.
+
+1. Open `alembic.ini` and update the `sqlalchemy.url` property to specify the correct connection string to your database:
+
+    For example:
+
+    ~~~
+    sqlalchemy.url = cockroachdb://demo:demo72529@127.0.0.1:26257/bank?sslmode=require
+    ~~~
+
+    {{site.data.alerts.callout_info}}
+    You must use the `cockroachdb://` prefix in the connection string for SQLAlchemy to make sure the CockroachDB dialect is used. Using the `postgres://` URL prefix to connect to your CockroachDB cluster will not work.
```

Comment:
Support for `postgres://` was deprecated years ago and finally [removed in SQLA 1.4](https://github.com/sqlalchemy/sqlalchemy/blob/e7119aea7870f0322e78d3a2cb28337b1640f0c2/doc/build/changelog/changelog_14.rst?plain=1#L4867). It's all `postgresql://` now.

Reply:
Got it. Updated.